### PR TITLE
Add FluxCD support for S3 Bucket sources

### PR DESCRIPTION
**Pull request by jkataja**
*Issue #, if available:*

*Description of changes:*
 - implement support for FluxCD Bucket source: https://fluxcd.io/flux/components/source/buckets/
 - bump FluxCD default version to 2.13.0
 - use the more reliable OCI registry for the FluxCD Helm chart
 - make indentation consistent within the modified files
 - changed typo in `defaultKustomiationProps`
 - basic use case tested
 - kustomizations not tested

Example use:

```
        .addOns(new blueprints.addons.FluxCDAddOn({
            buckets: [
                {
                    name: "bootstrap-bucket",
                    bucketName:base.bucket.bucketName,
                    bucketRegion: cdk.Aws.REGION,
                }
            ],
        }),
``` 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by elamaran11 — COMMENTED** [review]
Approach LGTM. @shapirov103 Pls check.

**Comment by jkataja** [comment]
@shapirov103 Hi, do you require any inputs from me to test and review the pull request? For example how I setup the Flux CD source bucket.

File: lib/addons/fluxcd/index.ts [hunk 1]
```diff
@@ -7,51 +7,102 @@ import { createNamespace, supportsALL } from "../../utils";
...
+     */
+    syncInterval?: string;
+
+    /**
+     * List of kustomizations to create from different paths in repo
+     */
+    kustomizations?: FluxKustomizationProps[];
+}
+
+/**
```
Comment (shapirov103) [inline lib/addons/fluxcd/index.ts:35]: As far as design - the`repository` construct is effectively misused as a vehicle to carry bucket info. Consider a case when we add a URL validator  (lack of it is a miss) to the repoUrl and start failing valid cases for the bucket name.
why not omit this structure from the type and introduce a clear bucket type?

File: lib/addons/fluxcd/index.ts [hunk 2]
```diff
@@ -60,53 +111,65 @@ export interface FluxKustomizationProps {
...
-  release: "blueprints-fluxcd-addon",
-  repository: "https://fluxcd-community.github.io/helm-charts",
-  values: {},
-  createNamespace: true,
+    name: "fluxcd-addon",
+    namespace: "flux-system",
+    chart: "flux2",
+    version: "2.13.0",
+    release: "blueprints-fluxcd-addon",
+    repository: "oci://ghcr.io/fluxcd-community/charts/flux2",
```
Comment (shapirov103) [inline lib/addons/fluxcd/index.ts:156]: with the switch to OCR, are you aware of a mechanism for us to discover when new versions of flux become available (other than manual look up)?

**Review by shapirov103 — COMMENTED** [review]
@jkataja thank you for this contribution, looks good overall, I added a couple of design/maintenance level comments. 

As far as testing, please share an example that you used for validation. We need to run it manually in this case as flux is not continuously tested with the end-to-end tests.

File: lib/addons/fluxcd/index.ts (hunk 2 above)
Comment (jkataja) [reply lib/addons/fluxcd/index.ts:160]: seems that `gh release` will show it but the output is not strictly for this chart:
```
$ gh release  list --repo fluxcd-community/helm-charts
flux2-sync-1.9.0           Latest  (flux2-sync-1.9.0)           about 1 month ago
flux2-notification-1.15.0          (flux2-notification-1.15.0)  about 1 month ago
flux2-2.13.0                       (flux2-2.13.0)               about 1 month ago
flux2-2.12.4                       (flux2-2.12.4)               about 3 months ago
flux2-sync-1.8.2                   (flux2-sync-1.8.2)           about 3 months ago
flux2-notification-1.14.1          (flux2-notification-1.14.1)  about 3 months ago
flux2-2.12.3                       (flux2-2.12.3)               about 3 months ago
flux2-notification-1.14.0          (flux2-notification-1.14.0)  about 4 months ago
flux2-sync-1.8.1                   (flux2-sync-1.8.1)           about 5 months ago
flux2-notification-1.13.1          (flux2-notification-1.13.1)  about 5 months ago
flux2-2.12.2                       (flux2-2.12.2)               about 5 months ago
flux2-2.12.1                       (flux2-2.12.1)               about 6 months ago
flux2-sync-1.8.0                   (flux2-sync-1.8.0)           about 6 months ago
flux2-notification-1.13.0          (flux2-notification-1.13.0)  about 6 months ago
flux2-2.12.0                       (flux2-2.12.0)               about 6 months ago
flux2-2.11.1                       (flux2-2.11.1)               about 7 months ago
flux2-sync-1.7.3                   (flux2-sync-1.7.3)           about 7 months ago
flux2-notification-1.12.4          (flux2-notification-1.12.4)  about 7 months ago
flux2-multi-tenancy-0.0.6          (flux2-multi-tenancy-0.0.6)  about 7 months ago
flux2-2.11.0                       (flux2-2.11.0)               about 7 months ago
flux2-2.10.6                       (flux2-2.10.6)               about 7 months ago
flux2-sync-1.7.2                   (flux2-sync-1.7.2)           about 7 months ago
flux2-notification-1.12.3          (flux2-notification-1.12.3)  about 7 months ago
flux2-2.10.5                       (flux2-2.10.5)               about 7 months ago
flux2-notification-1.12.2          (flux2-notification-1.12.2)  about 7 months ago
flux2-multi-tenancy-0.0.5          (flux2-multi-tenancy-0.0.5)  about 7 months ago
flux2-2.10.4                       (flux2-2.10.4)               about 7 months ago
flux2-2.10.3                       (flux2-2.10.3)               about 7 months ago
flux2-2.10.2                       (flux2-2.10.2)               about 8 months ago
flux2-sync-1.7.1                   (flux2-sync-1.7.1)           about 8 months ago
```
commented in 455a6a3fdf3a1bde30d42d232d306eb5a1bf2992

File: lib/addons/fluxcd/index.ts (hunk 1 above)
Comment (jkataja) [reply lib/addons/fluxcd/index.ts:35]: see commit 1c95b13078e8ea6406b7de009220763952bd5e0d

- [ ] test bucket construct changes

**Comment by jkataja** [comment]
> @jkataja thank you for this contribution, looks good overall, I added a couple of design/maintenance level comments.

You're welcome!

> As far as testing, please share an example that you used for validation. We need to run it manually in this case as flux is not continuously tested with the end-to-end tests.

Sure, so I create the bucket in a separate stack using:

```
new s3.Bucket(this, "SourceBucket", {
      removalPolicy: cdk.RemovalPolicy.RETAIN,
      objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_ENFORCED,
    });
```
Allow the node role to access the bucket by using:
```
        base.bucket.grantRead(role);
``` 

And then initialize the FluxCD addon as follows:
```
    .addOns(new blueprints.addons.FluxCDAddOn({
        buckets: [
            {
                name: "bootstrap-bucket",
                bucketName:base.bucket.bucketName,
                bucketRegion: cdk.Aws.REGION,
            }
        ],
    }),
...
``` 

Contains untested changes.

**Comment by jkataja** [comment]
The pipeline started failing with the eksworkshop.com returning HTTP 403 with AccessDenied response from S3

**Review by shapirov103 — APPROVED** [review]
LGTM

**Comment by elamaran11** [comment]
@jkataja Please merge from main, that should fix the GH Action errors.

**Comment by jkataja** [comment]
@elamaran11 I rebased on main, please try again

**Comment by elamaran11** [comment]
@shapirov103 LG to merge.

**Pull request merged**